core/src/test/java/org/apache/brooklyn/enricher/stock/PercentageEnricherTest.java

```diff
@@ -0,0 +1,275 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.enricher.stock;
+
+import static org.testng.Assert.assertEquals;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.brooklyn.api.entity.EntitySpec;
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.api.sensor.EnricherSpec;
+import org.apache.brooklyn.core.entity.BrooklynConfigKeys;
+import org.apache.brooklyn.core.entity.EntityAsserts;
+import org.apache.brooklyn.core.location.SimulatedLocation;
+import org.apache.brooklyn.core.sensor.BasicAttributeSensor;
+import org.apache.brooklyn.core.test.BrooklynAppUnitTestSupport;
+import org.apache.brooklyn.core.test.entity.TestApplication;
+
+public class PercentageEnricherTest extends BrooklynAppUnitTestSupport {
+
+    public static final Logger log = LoggerFactory.getLogger(PercentageEnricherTest.class);
+
+    AttributeSensor<Double> currentSensor;
+    AttributeSensor<Double> totalSensor;
+    AttributeSensor<Double> targetSensor;
+
+    @BeforeMethod(alwaysRun = true)
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        currentSensor = new BasicAttributeSensor<Double>(Double.class, "current");
+        totalSensor = new BasicAttributeSensor<Double>(Double.class, "total");
+        targetSensor = new BasicAttributeSensor<Double>(Double.class, "target");
+
+        app.start(ImmutableList.of(new SimulatedLocation()));
+    }
+
+    private void addEnricher() {
+        app.enrichers().add(EnricherSpec.create(PercentageEnricher.class)
+                .configure(PercentageEnricher.SOURCE_CURRENT_SENSOR, currentSensor)
+                .configure(PercentageEnricher.SOURCE_TOTAL_SENSOR, totalSensor)
+                .configure(PercentageEnricher.TARGET_SENSOR, targetSensor));
+    }
+
+    @Test
+    public void vanillaTest() {
+        addEnricher();
+
+        app.sensors().set(currentSensor, 50d);
+        app.sensors().set(totalSensor, 100d);
+        EntityAsserts.assertAttributeEqualsEventually(app, targetSensor, 50d);
+    }
+
+    @Test
+    public void currentNullTest() {
+        addEnricher();
+
+        app.sensors().set(currentSensor, null);
+        app.sensors().set(totalSensor, 100d);
+        EntityAsserts.assertAttributeEqualsEventually(app, targetSensor, null);
```

Comment:
Do you want to test that it set the value to null, or that it didn't change the pre-existing value? I think it will do the latter. If you want to test that, then you want to initialise it to something first. Same for other tests.